sentry-android/src/main/java/io/sentry/android/AndroidSentryClientFactory.java

```diff
@@ -91,6 +98,34 @@ public SentryClient createSentryClient(Dsn dsn) {
         SentryClient sentryClient = super.createSentryClient(dsn);
         sentryClient.addBuilderHelper(new AndroidEventBuilderHelper(ctx));
 
+        boolean enableAnrTracking = "true".equalsIgnoreCase(Lookup.lookup("anr.enable", dsn));
+        Log.d(TAG, "ANR is='" + String.valueOf(enableAnrTracking) + "'");
+        if (enableAnrTracking && anrWatchDog == null) {
```

Comment:
If `anrWatchDog` is supposed to be a singleton object then I think you should synchronize this block since multiple threads could potentially enter this critical path at the same time while `anrWatchDog` is still `null` and create multiple `ANRWatchDog` objects (i.e. start multiple threads).

Reply:
yes I guess I addressed this point on my reply above. Since this is the constructor of the `AndroidSentryClientFactory` and the usage of this type isn't by creating multiple instances, in multiple threads, the `volatile` should be good enough IMHO.

Reply:
Actually this is not the constructor of `AndroidSentryClientFactory` but the factory method (at first I thought the same, but we are inside `createSentryClient`). I see your point, still feels a half-way solution 😃

Reply:
right, create instead of constructor. Argument is the same though.